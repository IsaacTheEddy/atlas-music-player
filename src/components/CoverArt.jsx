import React from "react";
import cover from "../assets/placeholder.svg";

export function CoverArt() {
  return (
    <img
      className="size-100 flex-1 rounded-2xl"
      src={cover}
      alt="placeholder"
    />
  );
}
