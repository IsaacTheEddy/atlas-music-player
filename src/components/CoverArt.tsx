import { useCurrentSongContext } from "./context";
import placeholder from "../assets/placeholder.svg";
import { useState } from "react";

export default function CoverArt() {
  const { currentSong, loading } = useCurrentSongContext();
  return (
    <>
      {loading === true ? (
        <img
          className="size-100 flex-1 rounded-2xl"
          src="placeholder.svg"
          alt="placeholder"
        />
      ) : (
        <img
          className="size-100 flex-1 rounded-2xl"
          src={currentSong.cover}
          alt={currentSong.title}
        />
      )}
    </>
  );
}
