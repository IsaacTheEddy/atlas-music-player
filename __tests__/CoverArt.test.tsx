import { render } from "@testing-library/react";
import { test, expect } from "vitest";
import CoverArt from "../src/components/CoverArt";
import {
  currentSongContext,
  useCurrentSongContext,
} from "../src/components/context";

test("Pass data down through CoverArt", () => {
  let { container } = render(
    <currentSongContext.Provider
      value={{
        playlist: [],
        currentSong: {
          id: "a6pkp78whsyqdvpb5dxn64ss",
          title: "Fading Shadows",
          artist: "The Emberlight",
          genre: "Alternative Rock",
          duration: 194,
          cover:
            "https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL",
          song: "https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp",
          isPlaying: false,
        },
      }}
    >
      <CoverArt />
    </currentSongContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
test("Pass data down through CoverArt", () => {
  let { container } = render(
    <currentSongContext.Provider
      value={{
        playlist: [],
        currentSong: {
          id: "a6pkp78whsyqdvpb5dxn64ss",
          title: "Fading Shadows",
          artist: "The Emberlight",
          genre: "Alternative Rock",
          duration: 194,
          cover:
            "https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL",
          song: "https://utfs.io/f/E9fJnaKtTy1ba1N97yFrtlodEaJI0m4wGY9KgyPiUvCf8hMp",
          isPlaying: true,
        },
        loading: true,
      }}
    >
      <CoverArt />
    </currentSongContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
