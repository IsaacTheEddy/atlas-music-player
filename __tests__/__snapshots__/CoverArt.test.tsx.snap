// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image Loads correctly 1`] = `
<div>
  <img
    alt="Place"
    class="size-100 flex-1 rounded-2xl"
  />
</div>
`;
